fili-core/src/main/java/com/yahoo/bard/webservice/web/ResponseFormatResolver.java

```diff
@@ -0,0 +1,23 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web;
+
+import javax.ws.rs.container.ContainerRequestContext;
+
+/**
+ * A functional interface which allows the logic of resolving response format customizable. It has one method that
+ * takes the format name string from URI and a ContainerRequestContext object from which the header Accept field is
+ * accessible.
+ */
+public interface ResponseFormatResolver {
+    /**
+     * Resolve desirable format from URI and ContainerRequestContext. In case of a null return, Fili handles null
+     * format at {@link FiliResponseWriterSelector}, which use json as default value.
+     *
+     * @param format  The format String from URI
+     * @param containerRequestContext  ContainerRequestContext object that contains request related information
+     *
+     * @return A resolved format decided by the function
```

Comment:
The details on line 14-15 assumes things about where this will be called.  A function doesn't get to be that opinionated about it's use.

Instead add , null if none match to the @return clause.